cmake/modules/platform/toolcfg/gnu.cmake

```diff
@@ -161,14 +171,22 @@ function(_omr_toolchain_process_exports TARGET_NAME)
 
 	set(exp_file "$<TARGET_PROPERTY:${TARGET_NAME},BINARY_DIR>/${TARGET_NAME}.exp")
 
-	omr_process_template(
-		"${omr_SOURCE_DIR}/cmake/modules/platform/toolcfg/gnu_exports.exp.in"
-		"${exp_file}"
-	)
-
-if(NOT CMAKE_C_COMPILER_IS_OPENXL)
-	target_link_libraries(${TARGET_NAME}
-		PRIVATE
-			"-Wl,--version-script,${exp_file}")
-endif()
+	# Currently, OpenXL on AIX uses the GNU configuration. In the future, it should
+	# be switched to the OpenXL-specific configuration (as has already been done with
+	# z/OS), but until then, we need to add this special case for OpenXL
+	if (CMAKE_C_COMPILER_IS_OPENXL)
+		omr_process_template(
+			"${omr_SOURCE_DIR}/cmake/modules/platform/toolcfg/xlc_exports.exp.in"
+			"${exp_file}"
+		)
+		set_property(TARGET ${TARGET_NAME} APPEND_STRING PROPERTY LINK_FLAGS " -Wl,-bE:${TARGET_NAME}.exp")
```

Comment:
Why does this not use `target_link_libraries()` like below?
Or, perhaps better, `LINK_FLAGS` should be used in both cases?

Reply:
Switched the non-OpenXL configuration to use `LINK_FLAGS` instead.